src/parser.cr

```diff
@@ -59,7 +56,7 @@ module Mint
           input: data)
 
       part =
-        input[position, to].join
+        @string[position, to]
```

Comment:
If we store the original string we can compute substrings by calling `String#[]`, which is much faster than getting a subrange of `Array(Char)` and then joining that.

Maybe it's worth introducing a helper method for this? I don't see instance variables being used anywhere in the parser.

Reply:
I like having a helper method for this, not sure what to call it.

Reply:
I added one called `substring`